### PR TITLE
Grades endpoint

### PR DESCRIPTION
**Pull request by kaylambaker**
Adds endpoints /grades and /grades/all

/grades endpoint can query using fields from professors, sections, and courses. The query also has parameter representation which can be 'semester', 'section', or 'total'.
The query should be formatted as <collection>.<field>=<query>&representation=<semester|section|total>
example: /grades?professors.last_name=Alagar&representation=section

/grades/all returns data without query and has parameter representation which can be 'semester', 'section', or 'total'

### CODE REVIEW

File: api/server.go [hunk 1]
```diff
@@ -6,53 +6,63 @@ import (
...
-	if err != nil {
-		log.Print("There was no .env file found!")
-	}
-}
+	router := gin.Default()
 
-func loadMongo() {
-	configs.ConnectDB()
-}
+	// enable cors
```
Comment (ahshabbir) [inline api/server.go:18]: Should we remove this comment or re-enable cors?

File: api/server.go [hunk 2]
```diff
@@ -6,53 +6,63 @@ import (
...
+	//router.OPTIONS("*", CORSOptionsHandler())
+
+	// @DEBUG
+	// router.GET("/", func(c *gin.Context) {
+	//     c.String(http.StatusOK, "Hello World!")
+	// })
 
-	port, exist := os.LookupEnv("PORT")
-	if !exist {
+	port := os.Getenv("PORT")
```
Comment (ahshabbir) [inline api/server.go:49]: I think these lines are already executed by LookupEnv. If the environment variable doesn't exist, just set the port.

**Review by ahshabbir — CHANGES_REQUESTED** [review]
I think you also included your parser files. If you want them all in one PR, I would close the grade parser pr

**Comment by AdamMcAdamson** [comment]
I believe you updated the server.go erroneously, and included your parser code here too.
The only change there should have been the inclusion of ther line:
```golang
routes.GradesRoute(router)
```

The easiest thing to do would be to create a new branch and add the changes again, this time only including the changes specifically needed for the grades endpoint. (or you could reset this branch, and add the changes, but I don't want you to lose your changes..)

Otherwise, this looks good to me!

**Pull request closed without merging**